recipes/stlab/all/conanfile.py

```diff
@@ -0,0 +1,34 @@
+from conans import ConanFile, tools
+import os
+
+class Stlab(ConanFile):
+    name = 'stlab'
+    description = 'The Software Technology Lab libraries.'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/stlab/libraries'
+    license = 'BSL-1.0'
+    topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
+
+    settings = 'compiler'
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    requires = 'boost/1.69.0'
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "libraries-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        # TODO: Enable transitive required C++17
```

Comment:
You can obtain a good example here: https://github.com/conan-io/conan-center-index/blob/79541efd967e9a07704a96f63444681c3dbfaaf7/recipes/sol2/3.x.x/conanfile.py#L41

Reply:
Cool. But there's no profile in the CI that enables C++17 by default right? I would say that recipe compiles 0 packages in the CI (Which is what happened in my first build).

Reply:
Because self.settings.compiler.cppstd is None by default. That's why you need to check if it's available first by `if self.settings.get_safe("compiler.cppstd")`. Otherwise, tools.check_min_cppstd() will use the default c++ standard supported by the compiler version.

Reply:
But that is what `tools.check_min_cppstd(self, '17')` does, isn't it? It gets the `cppstd` value from your profile and checks if it is compatible with your requirement, else raises a `ConanInvalidConfiguration` exception.